go-get-modules-tests.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+set -e
+
+echo "Running go get tests with Go modules enabled"
+
+# Keep existing state
+export ORIG_PATH=$PATH
+export ORIG_GIT=$(which git)
+export ORIG_GOPATH=$GOPATH
+
+echo "Adding dgit to the path"
+go build
+mkdir -p bin
+cp dgit bin/git
+export PATH=$(pwd)/bin:$PATH
+
+export DGIT_TRACE=/tmp/go-get-dgit-log.$$.txt
+
+# Use a fresh GOPATH
+mkdir /tmp/gopath.$$
```

Comment:
Why not unset GOPATH to really make sure it works in module mode?

Reply:
I wanted to make sure that the modules get loaded into a fresh GOPATH separate from the one use to build dgit in the first place. The 'tests' check that those directories get created.

I wonder if GOPATH is going away with the modules or if it is being kept around. The documentation doesn't seem to say one way or another. I suspect that they'll need to keep it around for the module caches.

Reply:
You're definitely supposed to be able to build outside of $GOPATH in module mode, which is why I'd be more confident in is not being set than depending on an env which I think is only supposed to be a transitionary thing... if you're just concerned about polluting the real $GOPATH, how about setting it to a tmp dir like you are,  and then running the tests in a different tmp dir outside of both $GOPATHs?

Reply:
Ok, that sounds good. I've made the change to the script.

Reply:
Great, thanks for adding the test.. I'll merge as soon as it passes Travis